packages/command-bus/src/command-bus.ts

```diff
@@ -0,0 +1,20 @@
+type Type<T> = { new (...args: any[]): T }
+
+export class CommandBus<Context, Command> {
+  private handlers = new Map<Function, Handler<Context>>()
+
+  constructor(readonly target: Context) {}
+
+  handle(commandType: Type<Command>, handler: Handler<Context>) {
+    this.handlers.set(commandType, handler)
+  }
+
+  do(command: Command) {
+    const handler = this.handlers.get(command.constructor)
+    return handler(this.target, command)
+  }
+}
+
+export interface Handler<Context> {
+  (target: Context, command: any): unknown
```

Comment:
@jbpros this bit here is something I was still hoping to fix - the `any` type on the `command` parameter. It would make the handler functions neater to define if the compier already knew what type the command would be. What I found though, when I tried to type this, was that it wasn't happy to have all the different handlers in the map.

Worth a little play with this as a pair maybe?